hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteTableProcedure.java

```diff
@@ -100,6 +101,12 @@ protected Flow executeFromState(final MasterProcedureEnv env, DeleteTableState s
           // TODO: Move out... in the acquireLock()
           LOG.debug("Waiting for RIT for {}", this);
           regions = env.getAssignmentManager().getRegionStates().getRegionsOfTable(getTableName());
+          // We need to archive SPLIT regions as well, so add the SPLIT regions to the region list
+          List<RegionInfo> splitRegions = env.getAssignmentManager().getRegionStates()
```

Comment:
I think here we could just get all the regions of the table.

And look at the code of getRegionsOfTable(TableName, boolean), I'm afraid the implementation of the include method has something wrong?

```
    if (node.isInState(State.SPLIT) || hri.isSplit()) {
      return false;
    }
    if ((node.isInState(State.OFFLINE) || hri.isOffline()) && !offline) {
      return false;
    }
    return (!hri.isOffline() && !hri.isSplit()) ||
        ((hri.isOffline() || hri.isSplit()) && offline);
```

I do not think we need the two extra if conditions? Especially the first one, it will make us always exclude split parent?